### PR TITLE
Battery rack update

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Fixes battery racks not working by default
Adds quick-toggles to battery racks
Adds more examine text + the ability to remove batteries either by hand attack or by examine
Fixes code related to capacitor and battery insertion
Adds the new better and updates sprites
Also updates power generator sprites

## Why It's Good For The Game
Fix is always good

## Testing
Tested on local , built a powerrack from zero , tested insertion and removal , examine removal and hand removal , power charging and discharging.
Also tested the new generator sprites , made sure they update properly.
## Changelog
:cl:
fix: Fixed Battery racks not working
add: Added the abiltity to remove batteries from the battery racks (by examine or by hand attack)
add: Added visual status updates to the battery rack
add: Added more examine info to the battery racks.
add: Updated power generator sprites (taken from the #8380 
fix: Fixed battery racks not accepting new batteries / capacitors
balance: Battery racks no longer require 3 cells by default , they now require 1 capacitor and 20 cable coil
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
image
Ingame sprite

File: code/modules/power/batteryrack.dm [hunk 1]
```diff
@@ -1,19 +1,86 @@
...
+		to_chat(user, SPAN_NOTICE("It currently hosts [cells_amount] cells"))
+		if(distance <= 1)
+			to_chat(user, SPAN_NOTICE("Click any cell below to remove them from \the [src]"))
+			for(var/obj/item/cell/battery in component_parts)
+				to_chat(user, SPAN_NOTICE("<a href='?src=\ref[src];remove_cell_in_hand=\ref[battery];user=\ref[user]'>\icon[battery] [battery.name]</a>"))
+	else
+		to_chat(user, SPAN_NOTICE("The hatch needs to be opened with a screwdiver to interact with the cells inside!"))
+	to_chat(user, SPAN_NOTICE("It currently has [capacitors_amount] capacitor's installed"))
+	to_chat(user, SPAN_NOTICE("It has a LCD screen. Left side is for charging and right side for discharging. Green means operating, Yellow means not discharging/charging or no network."))
+	to_chat(user, SPAN_NOTICE("Can toggle input and output off with CtrlClick and Altclick. It is currently set to discharge a max of [output_level], and recharge by a max of [input_level]"))
```
Comment (hyperioo) [inline code/modules/power/batteryrack.dm:33]:
```suggestion
		to_chat(user, SPAN_NOTICE("It currently hosts [cells_amount] cells."))
		if(distance <= 1)
			to_chat(user, SPAN_NOTICE("Click any cell below to remove them from \the [src]:"))
			for(var/obj/item/cell/battery in component_parts)
				to_chat(user, SPAN_NOTICE("<a href='?src=\ref[src];remove_cell_in_hand=\ref[battery];user=\ref[user]'>\icon[battery] [battery.name]</a>"))
	else
		to_chat(user, SPAN_NOTICE("The hatch needs to be opened with a screwdriver to interact with the cells inside!"))
	to_chat(user, SPAN_NOTICE("It currently has [capacitors_amount] capacitors installed."))
	to_chat(user, SPAN_NOTICE("It has a LCD screen. Left side is for charging and right side for discharging. Green means operating. Yellow means not discharging/charging or no network."))
	to_chat(user, SPAN_NOTICE("Can toggle input and output on/off with CtrlClick and AltClick. It is currently set to discharge at a maximum rate of [output_level] W, and recharge at a maximum rate of [input_level] W."))
```

File: code/modules/power/batteryrack.dm [hunk 2]
```diff
@@ -1,19 +1,86 @@
...
+			// No funny HREF switching to grab anything in the world... or anything inside the machine.
+			// Don't use locate here , it searches by Type instead of reference
+			if(!component_parts.Find(battery) || !istype(battery))
+				return
+			if(target.incapacitated(INCAPACITATION_DEFAULT))
+				return
+			if(!Adjacent(target))
+				to_chat(target, SPAN_NOTICE("You are too far away from \the [src]"))
+				return
+			to_chat(target, SPAN_NOTICE("You remove \the [battery] from [src]"))
```
Comment (hyperioo) [inline code/modules/power/batteryrack.dm:72]:
```suggestion
				to_chat(target, SPAN_NOTICE("You are too far away from \the [src]."))
				return
			to_chat(target, SPAN_NOTICE("You remove \the [battery] from \the [src]."))
```

**Comment by Kurgis** [comment]
> add: Updated power generator sprites (taken from the https://github.com/discordia-space/CEV-Eris/pull/8380

There seems to of been an oversight, no modification was made to the power.dmi.

**Pull request merged**